cin_validator/cin_validator_class.py

```diff
@@ -10,18 +14,42 @@
 
 class CinValidationSession:
     def __init__(self, filename, ruleset, issue_id=None) -> None:
-        # TODO detect filetype xml/csv/zip. check if the directory is a folder.
-        fulltree = ET.parse(filename)
-        root = fulltree.getroot()
-        self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
-
         self.ruleset = ruleset
         self.issue_id = issue_id
+        self.filename = filename
 
+        self.file_type_checker()
         self.create_error_report_df()
         self.create_json_report()
         self.select_by_id()
 
+    def file_type_checker(self):
+        fn = self.filename.name
+        if os.path.isfile(fn):
+            extension = pathlib.Path(fn).suffix
+            if extension[:4] == ".xml":
+                fulltree = ET.parse(fn)
+                root = fulltree.getroot()
+                self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
+                print("got here!")
+            elif extension[:4] == ".csv":
+                print(".csv support coming soon!")
+            elif extension[:4] == ".zip":
+                zf = zipfile.ZipFile(fn)
+                dfs = {
+                    text_file.filename[:-4]: pd.read_csv(zf.open(text_file.filename))
+                    for text_file in zf.infolist()
+                    if text_file.filename.endswith(".csv")
```

Comment:
I like that you checked you added the check for csv. I also wouldn't expect an xml file to be in a zip here.
Should we flag it if an xml is included among the csvs?
or simply pick out only the csvs?
or consider the xml only (since all tables can be generated from it)?